sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -153,6 +153,18 @@ class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
     }
   }
 
+  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = {
+    try {
+      if (Option(loadTable(newIdent)).isDefined) throw new TableAlreadyExistsException(newIdent)
```

Comment:
I'd like to avoid using try catch as the control flow. How about
```
if (tableExists(newIdent)) throw ...
```

Reply:
Sounds good.